peekingduck/declarative_loader.py

```diff
@@ -180,12 +178,69 @@ def _edit_config(
                         f"Config for node {node_name} does not have the key: {key}"
                     )
                 else:
+                    if key == "detect_ids" and node_name in (
+                        "model.yolo",
+                        "model.efficientdet",
+                    ):
+                        key, value = self._change_labels_to_ids(node_name, key, value)
                     dict_orig[key] = value
                     self.logger.info(
                         f"Config for node {node_name} is updated to: '{key}': {value}"
                     )
         return dict_orig
 
+    def _change_labels_to_ids(
+        self, node_name: str, key: str, value: list
+    ) -> Tuple[str, list]:
+        """Process model.yolo or model.efficientdet detect_ids and check for labels to
+        be converted to ids, e.g. person to 0, car to 2
+
+        Args:
+            node_name (str): to determine if node is yolo or efficientdet as both
+                             lists of ids are different.
+            key (str): can only be "detect_ids", else error.
+            value (list): list of labels/ids for detection.
+                          If ids, do nothing.
+                          If labels, to be converted to ids.
+
+        Returns:
+            Tuple[str, list]: "detect_ids", list of sorted ids for object detection
+        """
+        assert key == "detect_ids", f"Key Error: expect detect_ids but got {key}"
+
+        map_labels_ids = self._load_mapping(node_name)
+        # Use set to eliminate duplicates and/or errors (which are mapped to zero)
+        obj_ids_set = {
+            x if isinstance(x, int) else map_labels_ids[x] if x in map_labels_ids else 0
+            for x in value
+        }
+        obj_ids_sorted_list = sorted(list(obj_ids_set))
+
+        return key, obj_ids_sorted_list
+
+    def _load_mapping(self, node_name: str) -> Dict[str, int]:
+        """Loads labels to ids mapping file for object detection models.
+        For efficientdet, the file is utils/mapping_efficientdet.txt
+        For yolo, the file is utils/mapping_yolo.txt
+
+        Args:
+            node_name (str): Must be either model.yolo or model.efficientdet
+
+        Returns:
+            Dict[str, int]: mapping of labels to object ids relevant to given model
+        """
+        assert node_name in (
+            "model.yolo",
+            "model.efficientdet",
```

Comment:
Similarly, is this necessary since a previous check has been done.

If it's necessary, to reduce maintenance burden, is it possible to keep a list of `model.<model_name>` in some txt or yaml file and read from that instead?

Reply:
I think this check is necessary, to ensure that if future models also have a `detect_ids` key, that this code is properly updated for them.